Code/Tools/FBuild/FBuildCore/BFF/BFFBooleanExpParser.cpp

```diff
@@ -0,0 +1,959 @@
+// BFFBooleanExpParser
+//------------------------------------------------------------------------------
+
+// Includes
+//------------------------------------------------------------------------------
+#include "BFFBooleanExpParser.h"
+
+#include "Tools/FBuild/FBuildCore/Error.h"
+#include <Tools/FBuild/FBuildCore/BFF/BFFKeywords.h>
+#include <Tools/FBuild/FBuildCore/BFF/BFFParser.h>
+#include <Tools/FBuild/FBuildCore/BFF/BFFStackFrame.h>
+#include <Tools/FBuild/FBuildCore/BFF/BFFVariable.h>
+#include "Tools/FBuild/FBuildCore/BFF/Tokenizer/BFFTokenRange.h"
+
+#include "Core/Strings/AStackString.h"
+
+// Operator
+// Including keyword `in` and `not in`
+//------------------------------------------------------------------------------
+enum class Operator
+{
+    OP_UNKNOWN,
+
+    OP_IN,
+    OP_NOT_IN,
+
+    OP_EQUAL,
+    OP_NOT_EQUAL,
+    OP_LESS_THAN,
+    OP_LESS_THAN_OR_EQUAL,
+    OP_GREATER_THAN,
+    OP_GREATER_THAN_OR_EQUAL,
+
+    OP_AND,
+    OP_OR,
+    // OP_XOR, not supported yet.
+};
+
+// GetVar
+//------------------------------------------------------------------------------
+static const BFFVariable * GetVar( const Function * function, const BFFToken * token )
+{
+    if ( token->IsVariable() == false )
+    {
+        Error::Error_1200_ExpectedVar( token, function );
+        return nullptr;
+    }
+
+    AString varName;
```

Comment:
This could use AStackString<> to avoid a heap allocation/free for every call of this function.

Reply:
Fixed in 98ef95275848f20cea086e8c02cc3e13c8657003.